src/main/java/org/broadinstitute/hellbender/tools/spark/utils/IntHistogram.java

```diff
@@ -198,6 +202,46 @@ public boolean isDifferentByKSStatistic( final IntHistogram sampleHistogram, fin
             return false;
         }
 
+        public KSData evalKS( final IntHistogram sampleHistogram ) {
+            final long[] sampleCounts = sampleHistogram.counts;
+            Utils.validateArg(sampleCounts.length == cdfFractions.length,
+                                "The supplied histogram doesn't have the right size.");
+            final long mCounts = sampleHistogram.getTotalObservations();
+            if ( mCounts == 0L ) return null;
+
+            int locationOfMaxDiff = -1;
+            float maxDiff = 0.f;
+            float maxArea = 0.f;
+            long sum = 0L; // sum from 0 to idx of sample histogram counts
+            int idx = 0; // current index into sample histogram
+            float curVal = 0.f; // sum/mCounts as a float (between 0 and 1)
+            float curArea = 0.f;
+            while ( sum < mCounts ) {
```

Comment:
In your previous implementation of this you had a check for "an extremum can occur at the end of a run of zero counts" but I don't see that in this version. Was it not longer needed?